src/pages/Trips/EditTrip.tsx

```diff
@@ -0,0 +1,245 @@
+    }, 5 * 1000);
+    return () => clearTimeout(timer);
+  };
+
+  const addDate = async () => {
+    setDates([...dates, emptyTripDate]);
+  };
+
+  const removeDate = async (index: number) => {
+    setDates(dates.filter((_, i) => i !== index));
+  };
+
+  const updateDate = async (updatedDate: TripDate, index: number) => {
+    const datesCopy = [...dates];
+    datesCopy[index] = updatedDate;
+    setDates(datesCopy);
+  };
+
+  const updateTrip = async () => {
+    setFailedUpdateFeedback(false);
+    setSuccessUpdateFeedback(false);
+    if (name === trip.name && dates == trip.dates && trip.private == privacy) {
+      setUpdateError('No fields are changed');
+      setFailedUpdateFeedback(true);
+      hideFailedFeedback();
+      return false;
+    }
+
+    const updatedTrip = {
+      name: name,
+      dates: dates,
+      private: privacy
+    } as Trip;
+
+    await api
+      ?.put(`/trips/${trip.id}`, updatedTrip)
+      .then((response) => {
+        setTrip((old) => [...old.filter((n) => n.id !== trip.id), response.trip]);
+        setSuccessUpdateFeedback(true);
+        hideSuccessFeedback();
+      })
+      .catch((response) => {
+        console.error('Not ok!');
+        console.debug(response.status, response.ok);
+        setUpdateError('Something failed, but who knows what?');
+        setFailedUpdateFeedback(true);
+        hideFailedFeedback();
+      });
+  };
+
+  return (
+    <Grid
+      container
+      direction="column"
+      justifyContent="center"
+      alignItems="center"
+      sx={{ minHeight: '100vh', backgroundColor: '#f5f5f5' }}>
+      <Grid size={{ sm: 12, md: 8, lg: 4 }}>
+        <Typography variant="h5" sx={{ mb: 2 }}>
+          Edit trip {trip.id}: {trip.name}
+        </Typography>
+      </Grid>
+      <Grid size={{ sm: 12, md: 8, lg: 4 }}>
+        <TextField
+          id="outlined-basic"
+          label={t('common.name')}
+          variant="outlined"
+          value={name}
+          onChange={(e) => setName(e?.target.value)}
+          sx={{ width: '100%' }}
+        />
+      </Grid>
+      <Grid size={{ sm: 12, md: 8, lg: 4 }}>
+        <Typography variant="h3" sx={{ mt: 2 }}>
+          {t('common.dates')}
+        </Typography>
+        {dates.map((date, index) => (
+          // Add a button to select the date
+          <Grid key={index} sx={{ mb: 1 }}>
+            <TripDateField
+              date={date}
+              updateDate={updateDate}
+              index={index}
+              onRemove={removeDate}
+            />
+          </Grid>
+        ))}
+        <IconButton aria-label="add" color="primary" sx={{ mt: 0.5 }} onClick={() => addDate()}>
+          <AddIcon />
+        </IconButton>
+      </Grid>
+      <Grid size={{ sm: 12, md: 8, lg: 4 }}>
+        <Stack direction="row" spacing={1} alignItems="center">
+          <Typography>{t('common.private')}</Typography>
+          <Switch value={privacy} onChange={() => setPrivacyNote(!privacy)} />
```

Comment:
Tror checked må settes på switchen for at den skal vise riktig verdi

```suggestion
          <Switch value={privacy} checked={!privacy} onChange={() => setPrivacyNote(!privacy)} />
```